processors/blockprocessor/block_processor.go

```diff
@@ -0,0 +1,367 @@
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/rpcs"
+
+	"github.com/algorand/indexer/accounting"
+	"github.com/algorand/indexer/data"
+	"github.com/algorand/indexer/plugins"
+	"github.com/algorand/indexer/processors"
+	indexerledger "github.com/algorand/indexer/processors/eval"
+	"github.com/algorand/indexer/util"
+)
+
+const implementationName = "block_evaluator"
+
+// BlockProcessor is the block processors interface
+type BlockProcessor interface {
+	NextRoundToProcess() uint64
+
+	processors.Processor
+}
+
+// blockprocessor package-wide init function
+func init() {
+	processors.RegisterProcessor(implementationName, &Constructor{})
+}
+
+type blockProcessor struct {
+	handler func(block *ledgercore.ValidatedBlock) error
+	ledger  *ledger.Ledger
+	logger  *log.Logger
+
+	cfg plugins.PluginConfig
+	ctx context.Context
+
+	// lastValidatedBlock is the last validated block that was made via the Process() function
+	// to be used with the OnComplete() function
+	lastValidatedBlock ledgercore.ValidatedBlock
+	// lastValidatedBlockRound is the round at which to add the last validated block
+	lastValidatedBlockRound       basics.Round
+	lastValidatedBlockCertificate agreement.Certificate
+}
+
+// Constructor is the ProcessorConstructor implementation for the "block processor" processor
+type Constructor struct{}
+
+// New initializes a blockProcessorConstructor
+func (c *Constructor) New() processors.Processor {
+	return &blockProcessor{}
+}
+
+func (proc *blockProcessor) Metadata() processors.ProcessorMetadata {
+	return processors.ProcessorMetadata{
+		ImplementationName:        implementationName,
+		ImplementationDescription: "Local Ledger Block Processor",
+		ImplementationDeprecated:  false,
+	}
+}
+
+func (proc *blockProcessor) Config() plugins.PluginConfig {
+	return proc.cfg
+}
+
+func (proc *blockProcessor) Init(ctx context.Context, initProvider data.InitProvider, cfg plugins.PluginConfig) error {
+
+	// First get the configuration from the string
+	var pCfg processors.BlockProcessorConfig
+	err := yaml.Unmarshal([]byte(cfg), &pCfg)
+
+	if err != nil {
+		return fmt.Errorf("blockprocessor init error: %w", err)
+	}
+
+	genesis := initProvider.Genesis()
+	round := uint64(initProvider.NextDBRound())
+
+	err = InitializeLedger(ctx, proc.logger, round, *genesis, &pCfg)
+	if err != nil {
+		return fmt.Errorf("could not initialize ledger: %w", err)
+	}
+
+	l, err := util.MakeLedger(proc.logger, false, genesis, pCfg.IndexerDatadir)
+	if err != nil {
+		return fmt.Errorf("could not make ledger: %w", err)
+	}
+	if l == nil {
+		return fmt.Errorf("ledger was created with nil pointer")
+	}
+
+	nextRound := initProvider.NextDBRound()
+	if uint64(l.Latest()) > uint64(nextRound) {
+		return fmt.Errorf("the ledger cache is ahead of the required round (%d > %d) and must be re-initialized", l.Latest(), nextRound)
+	}
```

Comment:
nit: you already have the `round` variable.
```suggestion
	if uint64(l.Latest()) > uint64(round) {
		return fmt.Errorf("the ledger cache is ahead of the required round (%d > %d) and must be re-initialized", l.Latest(), round)
	}
```